arches_component_lab/src/arches_component_lab/widgets/FileListWidget/components/FileListWidgetEditor/FileListWidgetEditor.vue

```diff
@@ -0,0 +1,136 @@
+<script setup lang="ts">
+import { onMounted, ref } from "vue";
+
+import FileUpload from "primevue/fileupload";
+
+import GenericFormField from "@/arches_component_lab/generic/GenericFormField.vue";
+import FileList from "@/arches_component_lab/widgets/FileListWidget/components/FileListWidgetEditor/components/FileList.vue";
+import FileDropZone from "@/arches_component_lab/widgets/FileListWidget/components/FileListWidgetEditor/components/FileDropZone.vue";
+
+import type { NodeData } from "@/arches_component_lab/types.ts";
+import type { FileReference } from "@/arches_component_lab/widgets/types.ts";
+import type { FileListCardXNodeXWidgetData } from "@/arches_component_lab/datatypes/file-list/types.ts";
+
+interface FileData {
+    name: string;
+    size: number;
+    type: string;
+    url: string;
+    file: File;
+    node_id: string;
+}
+
+type PrimeVueFile = File & { objectURL: string };
+
+const { value, nodeAlias, cardXNodeXWidgetData } = defineProps<{
+    value: NodeData | null | undefined;
+    nodeAlias: string;
+    cardXNodeXWidgetData: FileListCardXNodeXWidgetData;
+}>();
+
+const savedFiles = ref<FileReference[]>([]);
+const pendingFiles = ref<FileData[]>([]);
+
+const allowedFileTypes = ref();
+const currentValues = ref();
+
+onMounted(() => {
```

Comment:
funnily enough `watchEffect` doesn't work for this. Something to eventually look into